cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java

```diff
@@ -327,17 +341,22 @@ private List<CloudVmInstanceStatus> fillCloudVmInstanceStatuses(AuthenticatedCon
         return cloudVmInstanceStatuses;
     }
 
-    private Collection<String> instanceIdsWhichAreNotInCorrectState(List<CloudInstance> vms, AmazonEc2Client amazonEC2Client, String state) {
+    private ImmutablePair<Collection<String>, Collection<String>> instanceIdsWhichAreNotInCorrectState(List<CloudInstance> vms,
```

Comment:
This can be refactored. I don't think we should account for unfavourable states by using an `ImmutablePair`. You can determine pending instances by filtering from the original collection.

Reply:
Refactored it